### PR TITLE
feat(bearer-auth): added custom response message options

### PR DESCRIPTION
**Pull request by marceloverdijk**
### The author should do the following, if applicable

- [x] Add tests
- [x] Run tests
- [x] `bun run format:fix && bun run lint:fix` to format the code
- [x] Add [TSDoc](https://tsdoc.org/)/[JSDoc](https://jsdoc.app/about-getting-started) to document the code

### CODE REVIEW

**Comment by marceloverdijk** [comment]
@yusukebe here is the alternative PR: https://github.com/honojs/hono/pull/3381

File: src/middleware/bearer-auth/index.ts [hunk 1]
```diff
@@ -77,24 +97,48 @@ export const bearerAuth = (options: BearerAuthOptions): MiddlewareHandler => {
...
-        headers: {
-          'WWW-Authenticate': `${wwwAuthenticatePrefix}realm="` + realm + '"',
-        },
-      })
-      throw new HTTPException(401, { res })
+      const status = 401
+      const headers = {
+        'WWW-Authenticate': `${wwwAuthenticatePrefix}realm="` + realm + '"',
+      }
+      const responseMessage =
```
Comment (yusukebe) [inline src/middleware/bearer-auth/index.ts:104]: The logic with these lines appears three times in this file. Can we make them into one function with unified logic?

Comment (marceloverdijk) [reply src/middleware/bearer-auth/index.ts:104]: Absolutely, I will update the PR later today.

Comment (marceloverdijk) [reply src/middleware/bearer-auth/index.ts]: @yusukebe PR updated.

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
@marceloverdijk 

Thanks! Looks good to me. I'll merge this into the `next` later. Appreciate your contribution!

**Pull request merged**